fix: Allow ipv6 IPs to be admitted.

Kelketek opened the pull request:
## Description

This PR allows for IPv6 addresses to be permitted. It does so by swapping out to using the ipaddress module for validation, which is in all currently supported versions of Python. It should resolve #105 .

## Testing instructions

I haven't been able to run the tests because they seem dependent on now-unsupported 2.7 compatibility shims, and I ran out of time. Accordingly, I haven't been able to add any. I've attempted to run with 3.10 and with 3.8. Advice?

It does seem to be working for me in production, though.

bashu commented:
can you please update this PR, but this time without touching .gitignore, since it's unrelated to this issue

Kelketek commented:
@bashu Done!

Update: Whoops, accidentally committed one of the ignored files when I did that. Fixed now, though, and also removed some unneeded lines I forgot to prune earlier.

Kelketek committed:
fix: Allow ipv6 IPs to be admitted.
